### ISSUE TITLE
Apparent C# 8 bug in Range implementation

### ISSUE THREAD

**Issue by Korporal** [dotnet/roslyn#36122]
**C# 8 beta - Visual Studio 16.1.1**: 

**Steps to Reproduce**:

```cs
        public static T[] FromStart<T>(this T[] Source, int Count, int Offset = 0)
        {
            return Source[(Offset)..(Offset + Count)];
        }
```

**Expected Behavior**:

Compiler treats this the same as:

```cs
        public static T[] FromStart<T>(this T[] Source, int Count, int Offset = 0)
        {
            return Source[Offset..(Offset + Count)];
        }
```

**Actual Behavior**:

Complains and issues CS0118.

**Comment by CyrusNajmabadi** [comment]
@gafter I believe @Korporal is correct that this is a bug.  The compiler is treating this as if it's a cast expression.  However, the rules in the spec say this for disambiguating these cases

> To resolve cast_expression ambiguities, the following rule exists: A sequence of one or more tokens (White space) enclosed in parentheses is considered the start of a cast_expression only if at least one of the following are true:
>
> 1. ~Not applicable~
> 2. The sequence of tokens is correct grammar for a type, and the token immediately following the closing parentheses is the token "~", the token "!", the token "(", an identifier (Unicode character escape sequences), a literal (Literals), or any keyword (Keywords) except `as` and `is`.

In this case, the token immediately following is `..`, so this should not be considered a cast.

**Comment by agocke** [comment]
Should have been fixed in #36543
